spec-main/api-net-spec.ts

```diff
@@ -769,6 +774,59 @@ describe('net module', () => {
       })
     })
 
+    it('shold allow to read response body from non-2xx response', (done) => {
+      const bodyData = randomString(kOneKiloByte)
+      respondOnce.toSingleURL((request, response) => {
+        response.statusCode = 404
+        response.end(bodyData)
+      }).then(serverUrl => {
+        let requestResponseEventEmitted = false
+        let responseDataEventEmitted = false
+        let responseEndEventEmitted = false
+        let requestCloseEventEmitted = false
+
+        function maybeDone (done: () => void) {
+          if (!requestCloseEventEmitted || !responseEndEventEmitted) {
+            return
+          }
+
+          expect(requestResponseEventEmitted).to.equal(true)
```

Comment:
Calling `expect()` from an event / callback will result in the entire test process being aborted, instead of just failing the test. Please refactor similarly to https://github.com/electron/electron/pull/21265/files#diff-c31378831f4597a16ed943c2f7060323R566

Reply:
this is mostly copy-paste from existing test cases like https://github.com/electron/electron/blob/master/spec-main/api-net-spec.ts#L337 fyi

Reply:
yep, these changed with https://github.com/electron/electron/pull/21265